controller/contour/namespace.go

```diff
@@ -42,6 +42,12 @@ func (r *Reconciler) ensureNamespace(ctx context.Context, name string) error {
 		return nil
 	case errors.IsNotFound(err):
 		if err := r.Client.Create(context.TODO(), ns); err != nil {
+			// protect against a race condition between contour being updated
```

Comment:
not essential, but I think you could simplify this even further by not bothering to try getting the namespace first - just try creating it, and if you get an AlreadyExists error, you're good.